climpred/checks.py

```diff
@@ -143,3 +147,33 @@ def match_initialized_vars(init, ref):
             f'got {init_vars} for init and {ref_vars} for ref.'
         )
     return True
+
+
+def has_valid_lead_units(xobj):
+    """
+    Checks that the object has valid units for the lead dimension.
+    """
+    if hasattr(xobj['lead'], 'units'):
+
+        units = getattr(xobj['lead'], 'units')
+
+        # Check if letter s is appended to lead units string and add it if needed
+        if not units.endswith('s'):
+            units += 's'
+            xobj['lead'].attrs['units'] = units
+            warnings.warn(
+                f'The letter "s" was appended to the lead units; now {units}.'
+            )
+
+        # Raise Error if lead units is not valid
+        if not getattr(xobj['lead'], 'units') in VALID_LEAD_UNITS:
+            raise DimensionError(
+                'The lead dimension must must have a valid '
+                f'units attribute. Valid options are: {VALID_LEAD_UNITS}'
+            )
+    else:
+        raise DimensionError(
+            'The lead dimension must must have a '
```

Comment:
Did we have an exception.py module? This seems repeated

Reply:
The exception module just defines our custom errors like `DimensionError` and so on. Maybe we could have an error module that uses them for repeating errors like this?